lang/java/reef-bridge-java/src/main/java/org/apache/reef/javabridge/LibLoader.java

```diff
@@ -62,92 +62,58 @@ public void loadLib() throws IOException {
       this.loadBridgeDLL();
 
       // Load all DLLs in local
-      this.loadAllManagedDLLs(this.reefFileNames.getLocalFolder());
+      loadAllManagedDLLs(this.reefFileNames.getLocalFolder());
 
       // Load all DLLs in global
-      this.loadAllManagedDLLs(this.reefFileNames.getGlobalFolder());
+      loadAllManagedDLLs(this.reefFileNames.getGlobalFolder());
     }
     LOG.log(Level.INFO, "Done loading DLLs for Driver at time {0}." + new Date().toString());
   }
 
   /**
-   * Loads the Bridge DLL. First, it attempts to load from the reef/local folder. Second attempt is reef/global, last
-   * attempt is loading it from the JAR.
+   * Loads the Bridge DLL.
+   * <p>
+   * If the file is found in the reef/local folder, it is used. Else, we load teh one in reef/global. If that isn't
+   * present, this method throws an IOException
    *
-   * @throws IOException If all attempts fail.
+   * @throws FileNotFoundException if neither file is available.
+   * @throws Throwable             if the DLL is found, but System.load() throws an exception.
    */
-  private void loadBridgeDLL() throws IOException {
-    try {
-      loadBridgeDLLFromLocal();
-    } catch (final Throwable t) {
-      LOG.log(Level.INFO, "Unable to load bridge DLL from local folder. Attempting global folder next.", t);
-      try {
-        loadBridgeDLLFromGlobal();
-      } catch (final Throwable t2) {
-        LOG.log(Level.WARNING, "Unable to load bridge DLL from global folder. Attempting jar next.", t2);
-        loadBridgeDLLFromJAR();
-      }
+  private void loadBridgeDLL() throws FileNotFoundException {
```

Comment:
This method can accept File as a parameter. At caller side, you can call it twice, one with eefFileNames.getBridgeDLLInLocalFolderFile(), the other with reefFileNames.getBridgeDLLInGlobalFolderFile() so that you don't need to repeat the file exists logic for both local and global inside this method. 
Then, it actually can be consolidated with loadBridgeDLLFromFile(File) method.